### PR TITLE
Update ASP.NET existing db tutorial for .NET Core 2.1

### LINKED ISSUE #717: Include code for Post class in code snippets in tutorial
By given only the code of entity class Blog.cs, I try this tutorial but getting unwanted results. Please add Post.cs then it would be easier to realize the relation.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 998197fe-077b-acee-c91c-12f074359e9e
* Version Independent ID: 7542e87e-8f99-9b91-6df1-6983776ad1d2
* Content: [Getting Started on ASP.NET Core - Existing Database - EF Core](https://docs.microsoft.com/en-us/ef/core/get-started/aspnetcore/existing-db)
* Content Source: [entity-framework/core/get-started/aspnetcore/existing-db.md](https://github.com/aspnet/EntityFramework.Docs/blob/master/entity-framework/core/get-started/aspnetcore/existing-db.md)
* Service: **unspecified**
* Product: **entity-framework**
* GitHub Login: @rowanmiller
* Microsoft Alias: **divega**

### LINKED ISSUE #888: Update ASP.NET Core existing db tutorial for .NET Core 2.1
Update the tutorial and add enhancements suggested in other feedback issues.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 998197fe-077b-acee-c91c-12f074359e9e
* Version Independent ID: 7542e87e-8f99-9b91-6df1-6983776ad1d2
* Content: [Getting Started on ASP.NET Core - Existing Database - EF Core](https://docs.microsoft.com/en-us/ef/core/get-started/aspnetcore/existing-db)
* Content Source: [entity-framework/core/get-started/aspnetcore/existing-db.md](https://github.com/aspnet/EntityFramework.Docs/blob/master/entity-framework/core/get-started/aspnetcore/existing-db.md)
* Product: **entity-framework**
* GitHub Login: @rowanmiller
* Microsoft Alias: **divega**

### PR DESCRIPTION
**Pull request by tdykstra**
@divega -
* The sample app is updated to what the template + tutorial creates.
* Per the feedback issues:
  * I added the Post entity class in the doc. 
  * Also added a note about making nav properties virtual for lazy loading, which we can remove when 2.2 is released.
* The original tutorial deletes OnConfiguring, but since it now has `if (!optionsBuilder.IsConfigured)`  I left it in -- we could delete it anyway.
* At some point we should probably make this a Razor Pages tutorial; I could do that now if you think it's worth some extra time.

cc: @Rick-Anderson 

Fixes #888
Fixes #717 

[Review link](https://review.docs.microsoft.com/en-us/ef/core/get-started/aspnetcore/existing-db?branch=pr-en-us-887)

### CODE REVIEW

File: samples/core/GetStarted/AspNetCore/EFGetStarted.AspNetCore.ExistingDb/Controllers/BlogsController.cs [hunk 1]
```diff
@@ -124,7 +124,7 @@ public async Task<IActionResult> Delete(int? id)
             }
 
             var blog = await _context.Blog
-                .SingleOrDefaultAsync(m => m.BlogId == id);
+                .FirstOrDefaultAsync(m => m.BlogId == id);
```
[the remark this answers is not in the document]
Comment (tdykstra) [reply samples/core/GetStarted/AspNetCore/EFGetStarted.AspNetCore.ExistingDb/Controllers/BlogsController.cs]: This is what scaffolding put there and should be the same as what we have for Details and Edit. Not clear to me why Details and Delete have this while Edit has Find. I'll ask.

**Comment by Rick-Anderson** [comment]
The scaffolder uses `FirstOrDefaultAsync` everywhere except `POST Delete` where it uses `FindAsync`. This is what the EF team provided to me [FindAsync](https://docs.microsoft.com/en-us/aspnet/core/data/ef-rp/crud?view=aspnetcore-2.1#findasync)

@prafullbhosale might have a comment on the scaffolder

**Comment by tdykstra** [comment]
I'm seeing Find on Edit Get as well as Delete Post.

That RP tutorial section has this:

>But if you want to Include other entities, then FindAsync is no longer appropriate. This means that you may need to abandon FindAsync and move to a query as your app progresses.

For 2.1 it might be a good idea to mention lazy loading as an alternative there.

File: samples/core/GetStarted/AspNetCore/EFGetStarted.AspNetCore.ExistingDb/Controllers/BlogsController.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (divega) [reply samples/core/GetStarted/AspNetCore/EFGetStarted.AspNetCore.ExistingDb/Controllers/BlogsController.cs]: I didn’t realize I was looking at scaffolded code :smile:

**Pull request merged**